src/object/cli_obj.c

```diff
@@ -3525,14 +3527,14 @@ obj_comp_cb(tse_task_t *task, void *data)
 		 * shard, such as -DER_TIMEDOUT or daos_crt_network_error().
 		 */
 
-		if ((!obj_auxi->spec_shard && !obj_auxi->no_retry) ||
+		if ((!obj_auxi->spec_shard && !obj_auxi->spec_group) ||
```

Comment:
why remove no_retry here?

Reply:
because we will set "no_retry" if "!no_retry", it is not a big deal if we directly set it without checking it firstly.